pkg/controller/storagecluster/uninstall_reconciler.go

```diff
@@ -373,12 +373,12 @@ func (r *ReconcileStorageCluster) deleteResources(sc *ocsv1.StorageCluster, reqL
 		return err
 	}
 
-	err = r.deleteSnapshotClasses(sc, reqLogger)
+	err = ensureDeleted(&ocsSnapshotClass{}, r, sc, reqLogger)
```

Comment:
@iamniting , @jarrpa , we can reduce the whole `deleteResources()` function with two loops. Something like,
```go
	type resourceDeleteFunc func(*ocsv1.StorageCluster, logr.Logger) error

	allDeleteFuncs := []resourceDeleteFunc{
		r.setRookUninstallandCleanupPolicy,
		r.setNoobaaUninstallMode,
		r.deleteNoobaaSystems,
		r.deleteCephCluster,
		r.deleteCephObjectStoreUsers,
		r.deleteCephObjectStores,
		r.deleteCephFilesystems,
		r.deleteCephBlockPools,
		r.deleteNodeTaint,
	}

	allEstablishers := []establisher{
		&ocsSnapshotClass{},
		&ocsStorageClass{},
	}

	for _, eachDelFunc := range allDeleteFuncs {
		err := eachDelFunc(sc, reqLogger)
		if err != nil {
			return err
		}
	}

	for _, eachEstablisher := range allEstablishers {
		err := ensureDeleted(eachEstablisher, r, sc, reqLogger)
		if err != nil {
			return err
		}
	}

```
PS: please note that `r.deleteNodeTaint()` function is called after `ensureDeleted()` function but in the above suggestion it will be called prior to the `ensureDeleted()` functions

Reply:
This will be also taken care of in the last commit once we finalize it.

Reply:
Done